sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,574 @@
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should skip the ECE disbursement when the enrolment is already completed.", async () => {
+      // Arrange
+      const confirmEnrolmentInstitutionCode = "CONF";
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        confirmEnrolmentInstitutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+      // Mock the implementation to return the mocked institution code.
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
+        async () => {
+          return [confirmEnrolmentInstitutionCode];
+        };
+
+      // Create disbursement to confirm enrolment.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        undefined,
+        {
+          applicationStatus: ApplicationStatus.Completed,
+        },
+      );
+
+      const [disbursement] =
+        application.currentAssessment.disbursementSchedules;
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have the correct values for
+      // disbursement and application number.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent
+            .replace("DISBNUMBER", disbursement.id.toString().padStart(10, "0"))
+            .replace("APPLNUMBER", application.applicationNumber)
+            .replace("ENRLDATE", formatDate(new Date(), "YYYYMMDD"));
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        "Total disbursements found: 1",
+        "Disbursements successfully updated: 0",
+        "Disbursements skipped to be processed: 0",
+        "Disbursements considered duplicate and skipped: 1",
+        "Disbursements failed to process: 0",
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expectedResult.warnings = [
+        `Disbursement ${disbursement.id}, record is considered as duplicate and skipped due to reason: Enrolment already completed and can neither be confirmed nor declined`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should skip the ECE disbursement when disbursement and disbursement does not belong to the system.", async () => {
```

Comment:
`when the application doesn't belong to the system`?

Reply:
When it belongs to SFAS. The file can contain records from both systems.

Reply:
Actually my comment regarding the description, it is a saying ` when disbursement and disbursement does not belong to` . shouldn't it be just ` when disbursement  does not belong to  `